### PR TITLE
Managed mirrored speak animations in ESCItem

### PR DESCRIPTION
**Pull request by StraToN**
Fixes https://github.com/godot-escoria/escoria-issues/issues/155

cc @balloonpopper

### CODE REVIEW

File: addons/escoria-core/game/core-scripts/esc_item.gd [hunk 1]
```diff
@@ -482,6 +493,14 @@ func stop_talking():
 			and _movable.last_dir < animations.speaks.size():
 		if get_animation_player().is_playing():
 			get_animation_player().stop()
+		
+		
+		var talk_pose_scale = -1 \
+				if animations.speaks[_movable.last_dir].mirrored else 1
+
+		if talk_pose_scale == -1:
```
Comment (dploeger) [inline addons/escoria-core/game/core-scripts/esc_item.gd:501]: No need for the conditional. (see above)

**Review by dploeger — CHANGES_REQUESTED** [review]

**Comment by BHSDuncan** [comment]
Will review once changes are made.

File: addons/escoria-core/game/core-scripts/esc_item.gd [hunk 2]
```diff
@@ -468,6 +468,17 @@ func start_talking():
 			and _movable.last_dir < animations.speaks.size():
 		if get_animation_player().is_playing():
 			get_animation_player().stop()
+
+		var talk_pose_scale = -1 \
+				if animations.speaks[_movable.last_dir].mirrored else 1
+
+		# Do not flip the entire character, because that would conflict
```
[the remark this answers is not in the document]
Comment (StraToN) [reply addons/escoria-core/game/core-scripts/esc_item.gd:475]: That was because I used the same block of code as in `ESCMovable`:
https://github.com/godot-escoria/escoria-demo-game/blob/7aca986cf7e5e316876f49e972859030284c64d3/addons/escoria-core/game/core-scripts/behaviors/esc_movable.gd#L341-L348

[the remark this answers is not in the document]
Comment (StraToN) [reply addons/escoria-core/game/core-scripts/esc_item.gd]: Fixed

**Comment by dploeger** [comment]
@StraToN As I was reviewing this again I thought about whether we should change `pose_scale` to `is_mirrored` as a boolean, because it would make much more sense IMHO. What do you think?

**Comment by dploeger** [comment]
Did you forget to push the change? I still see `pose_scale`

File: addons/escoria-core/game/core-scripts/behaviors/esc_movable.gd [hunk 3]
```diff
@@ -169,8 +169,8 @@ func _perform_walk_orientation(angle: float):
 			true
 		)
 
-	pose_scale = -1 if parent.animations.directions[last_dir].mirrored \
-		else 1
+	is_mirrored = true if parent.animations.directions[last_dir].mirrored \
```
Comment (dploeger) [inline addons/escoria-core/game/core-scripts/behaviors/esc_movable.gd:172]: Another simplification: just write `is_mirrored = parent.animations.directions[last_dir].mirrored`

😃

**Review by dploeger — CHANGES_REQUESTED** [review]

File: addons/escoria-core/game/core-scripts/behaviors/esc_movable.gd (hunk 3 above)
Comment (StraToN) [reply addons/escoria-core/game/core-scripts/behaviors/esc_movable.gd:172]: Indeed, I was tired I think :D Will fix

**Review by dploeger — APPROVED** [review]
Great. Thanks. 🎁

**Comment by StraToN** [comment]
@BHSDuncan I have checked the necessity to add null checks but I think it's not useful at the moment. We have to assume that the ESCAnimationResource is correctly set up by gamedev, ie. animations filled for every created direction.
We will be able to ensure this by providing a visual tool in the plugin to create that resource.
TLDR: no new modification to this PR, please review :)

**Comment by BHSDuncan** [comment]
> @BHSDuncan I have checked the necessity to add null checks but I think it's not useful at the moment. We have to assume that the ESCAnimationResource is correctly set up by gamedev, ie. animations filled for every created direction. We will be able to ensure this by providing a visual tool in the plugin to create that resource. TLDR: no new modification to this PR, please review :)

I'm a fan of "defence in depth" but I also agree there has to be a limit.

**Review by BHSDuncan — APPROVED** [review]
